File: primes.cs
using System;

namespace dockerizecsharp
{
	class MainClass
	{
		public static void Main (string[] args)
		{
			Console.WriteLine ("C#");
			int minprime = 0, maxprime = 20000000, num, count = 0;
			var watch = System.Diagnostics.Stopwatch.StartNew();
			for (num = minprime; num <= maxprime; num++)
				if (isPrime(num)) count++;
			watch.Stop();
			Console.WriteLine ("Number of primes in the interval [{0},{1}]: {2}",minprime,maxprime,count);
			Console.WriteLine ("Time elapsed: {0} seconds",watch.ElapsedMilliseconds/1000.0);
		}
		
		static bool isPrime(int n)
		{
			if (n <= 1) return false;
			if (n <= 3) return true;
			if (((n % 2) == 0) || ((n % 3) == 0)) return false;
			for (int i = 5; i*i <= n; i += 6)            
				if (((n % i) == 0) || ((n % (i+2)) == 0)) return false;
			return true;
		}
		
	}
}

File: primes.fs
let isPrime n =
    if (n <= 1) then false
    elif (n<= 3) then true
    elif (((n % 2) = 0) || ((n % 3) = 0)) then false
    else
        let rec checkPrime i = 
            if i*i > n then true
            else
                if (((n % i) = 0) || ((n % (i+2)) = 0)) then false
                else checkPrime (i + 6)     
        checkPrime 5
        
[<EntryPoint>]
let main argv =
    
    printfn "F#"
    let minprime = 0
    let maxprime = 20000000
    let mutable count = 0

    let watch = System.Diagnostics.Stopwatch.StartNew();
    
    [minprime..maxprime] |> List.iter (fun num -> if isPrime num then count <- count + 1)

    watch.Stop();
    printfn "Number of primes in the interval [%d,%d]: %d" minprime maxprime count
    let elapsed = (float watch.ElapsedMilliseconds) / 1000.0
    printfn "Time elapsed: %f seconds" elapsed

    System.Console.ReadKey() |> ignore
    0
